doc/pmon/sonic_thermal_control_test_plan.md

```diff
@@ -0,0 +1,213 @@
+            ],
+            "actions": [
+                {
+                    "type": "thermal_control.control",
+                    "status": "true"
+                },
+                {
+                    "type": "fan.all.set_speed",
+                    "speed": "60"
+                }
+            ]
+        }
+    ]
+}
+```
+
+#### Invalid format policy file
+
+In invalid_format_policy.json, the file content is not valid JSON at all. A file with content "invalid" should be good for test purpose.
+
+#### Invalid value policy file
+
+In invalid_value_policy.json, the file content contains minus value of target speed. We couldn't cover all invalid value here because there are too many possibilities. The major purpose of this configuration file is to verify thermal control daemon would not crash while loading such an invalid configuration files. For other negative test cases, they are already covered by unit test. The invalid_value_policy.json content is like:
+
+```json
+{
+    "info_types": [
+        {
+            "type": "fan_info"
+        },
+        {
+            "type": "psu_info"
+        }
+    ],
+    "policies": [
+        {
+            "name": "any PSU absence",
+            "conditions": [
+                {
+                    "type": "fan.any.absence"
+                }
+            ],
+            "actions": [
+                {
+                    "type": "fan.all.set_speed",
+                    "speed": "-1"
+                }
+            ]
+        }
+    ]
+}
+```
+
+## Test Cases
+
+### Show FAN Status Test
+
+Show FAN status test verifies that all FAN related information can be shown correctly via `show platform fanstatus`.
+
+#### Procedure
+
+1. Testbed setup.
+2. Issue command `show platform fanstatus`.
+3. Read the command output.
+4. Read FAN status via sysfs.
+5. Verify that command output matches FAN status from sysfs.
+
+> Note: FAN status monitor update FAN status every 60 seconds, so the command output might not match FAN status from sysfs exactly. We may introduce some tolerance.
+
+### Show Thermal Status Test
+
+Show thermal status test verifies that all thermal related information can be shown correctly via `show platform temperature`.
+
+#### Procedure
+
+1. Testbed setup
+2. Issue command `show platform temperature`.
+3. Record the command output
+4. Read thermal status via sysfs.
+5. Verify that command output matches thermal status from sysfs.
+
+> Note: Thermal status monitor update FAN status every 60 seconds, so the command output might not match FAN status from sysfs exactly. We may introduce some tolerance.
+
+### PSU Absence Test
+
+PSU absence test verifies that once any PSU absence, all FAN speed will be set to proper value according to policy file.
+
+#### Procedure
+
+1. Testbed setup.
+2. Copy valid_policy.json to pmon docker and backup the original one.
+3. Restart pmon service to trigger thermal control daemon reload policy configuration file.
+4. Stop two PSUs.
```

Comment:
do you mean to power off the switch? 😀
regarding psu operations, I think you can refer the logic in PSU test cases which is something like:

    check whether all PSUs are turned on and exit if not(we don't want to turn off switch in this case)
    power off one PSU and check... 
    power on the PSU and power off the other and check... 
    finish

Reply:
Yes, I saw there is a psu_controller which can turn on/off the PSU.